dashboard/src/components/Header/ContextSelector.tsx

```diff
@@ -0,0 +1,134 @@
+  const dispatch = useDispatch();
+  const [open, setOpen] = useState(false);
+  const currentCluster = clusters.clusters[clusters.currentCluster];
+  const namespaceSelected = currentCluster.currentNamespace || defaultNamespace;
+  const [cluster, setCluster] = useState(clusters.currentCluster);
+  const [namespace, setNS] = useState(namespaceSelected);
+  // Control when users click outside
+  const ref = useRef(null);
+  useOutsideClick(setOpen, [ref], open);
+
+  useEffect(() => {
+    fetchNamespaces();
+    if (namespaceSelected !== definedNamespaces.all) {
+      getNamespace(namespaceSelected);
+    }
+  }, [fetchNamespaces, namespaceSelected, getNamespace]);
+
+  const toggleOpen = () => setOpen(!open);
+  const selectCluster = (event: React.ChangeEvent<HTMLSelectElement>) =>
+    setCluster(event.target.value);
+  const selectNamespace = (event: React.ChangeEvent<HTMLSelectElement>) =>
+    setNS(event.target.value);
+  const changeContext = () => {
+    setNamespace(namespace);
+    dispatch(push(app.apps.list(cluster, namespace)));
+    setOpen(false);
+  };
+
+  return (
+    <>
+      <div
+        className={`dropdown kubeapps-align-center kubeapps-dropdown ${open ? "open" : ""}`}
+        ref={ref}
+      >
+        <button className="kubeapps-nav-link" onClick={toggleOpen} aria-expanded={open}>
+          <Row>
+            <Column span={10}>
+              <span>Current Context</span>
+              <div>
+                <CdsIcon size="sm" shape="cluster" inverse={true} />
+                <span className="kubeapps-dropdown-text">{clusters.currentCluster}</span>
+                <CdsIcon size="sm" shape="file-group" inverse={true} />
+                <span className="kubeapps-dropdown-text">{namespaceSelected}</span>
+              </div>
+            </Column>
+            <Column span={2}>
+              <div
+                className={`kubeapps-align-center angle ${open ? "angle-opened" : "angle-closed"}`}
+              >
+                <CdsIcon shape="angle" inverse={true} />
+              </div>
+            </Column>
+          </Row>
+        </button>
+        <div className="dropdown-menu" role="menu">
+          <span className="context-selector-header dropdown-menu-padding">
+            Select a cluster and a namespace to manage applications
+          </span>
+          <div className="dropdown-menu-padding" role="menuitem">
+            <CdsIcon size="sm" shape="cluster" inverse={true} />
+            <span className="kubeapps-dropdown-text">Cluster</span>
+            <div className="clr-select-wrapper">
+              <select name="clusters" className="clr-page-size-select" onChange={selectCluster}>
+                {Object.keys(clusters.clusters).map(c => {
+                  return (
+                    <option key={`kubeapps-dropdown-cluster-${c}`} value={c}>
+                      {c}
+                    </option>
+                  );
+                })}
+              </select>
```

Comment:
The style of the selector options is not the correct one. Clarity lefts the browser to apply their own style on those selectors:

![Screenshot on how Clarity left selectors to use their native style based on the browser](https://user-images.githubusercontent.com/4056725/87011359-177e6200-c1c8-11ea-996d-63566277d21e.png)

Reply:
I don't understand what you are proposing, maybe it's a confusion? Note that I'm in linux so the style here is different:
![Screenshot from 2020-07-09 17-20-37](https://user-images.githubusercontent.com/4025665/87058568-87acd800-c208-11ea-960e-5dd757daf1d4.png)

Reply:
Ohh I see! Completely forgot that